### DISCUSSION TITLE
Script buffer size

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#16238] [Q&A] [answered]
@gemu2015 I'm working with a BMS script (which is working great, thanks to your SPI help). It's getting large, and I'm running into issues with buffer space. I'm reading through your documentation on the external editor but see some confusing info:

In your PDF doc you say:
```
on an ESP8266 you have at least 2 options for script buffer with standard 1M flash:
2560 bytes and compressed 1535 bytes (default configuration) SB=2560:1535
or if you have 4M Flash:
SB=4096
and in user_config_override:
#define
USE_UFILESYS
and select a different linker file in platform_override 
board_build.ldscript =
; eagle.flash.1m.ld
eagle.flash.4m2m.ld

on an ESP32 besides the default option you may use e.g.
SB=8192
and in user_config_override:
#define
USE_UFILESYS
#define
UFSYS_SIZE
8192
```
But in the scripting documentation it's written:
```
Feature	ESP8266	ESP32	PVARS	remarks
fallback	1536	1536	50	no longer supported
compression (default)	2560	2560	50	actual compression rate may vary
#define USE_UFILESYS
#define UFSYS_SIZE S	S<=8192	S<=16384	1536	ESP8266 must use 4M Flash use linker option -Wl,-Teagle.flash.4m2m.ld or SDCARD 
ESP32 can use any linker file, size of Filesystem depends on linker file
```
Suggesting that ESP8266 can handle a buffer size of 8192 with a 4MB flash chip.
So I need to compile with 
```
#define UFSYS_SIZE 8192
#define USE_UFILESYS
```
And then also type in `SB=8192` in external editor?

Second question: I need more than 5 arrays to store all the registers, which I see can be increased by `#define MAXFILT X` If I do this, my only concern is increased RAM usage? Or will it break something else to have, for example, 8 arrays?

**Comment by gemu2015** [comment, marked as answer]
according to script size on ESP8266 i can not recommend more then about 6,5 kB
and you should exclude any unused driver from the compilation to save additional RAM.

according to arrays the RAM usage depends on the actual defined array sizes not on the number of allowed arrays.
since all numbers are single precision float an array uses SIZE*4 bytes.

BTW  the last dev version allows a much shorter array initialization to save script space

\>B
array[0]=0
array={1 2 3 4}

**Reply by pkkrusty** [reply]
Yes, I saw that in docs, and thought I just missed it.  You also write about the option of using 
```
#define EEP_SCRIPT_SIZE 6200
#define USE_EEPROM
```
Do you have a preference between that and 
```
#define USE_UFILESYS
#define UFSYS_SIZE 4096
```
Is one more reliable than the other?

**Reply by gemu2015** [reply]
the upper is meant for 1M flash devices and has a few drawbacks and does NOT work on 4M devices.

on 4M devices use the file system with up to 6,5k script buffer
